### DISCUSSION TITLE
Scoop eclipse packages

### DISCUSSION THREAD

**Discussion by finrod2002** [ScoopInstaller/Extras#11880] [General]
Hello,

What is the recommended practice to handle eclipse packages installed via scoop. Currently, if I update eclipse (e.g. eclipse-jee) with scoop I will lose all installed extensions and plugins.

Should I use "scoop hold" and only use the eclipse auto updater instead of updating via scoop?

**Comment by rashil2000** [comment]
Do you know where the extensions and plugins are stored? We can persist them through updates by modifying the manifest.

**Comment by finrod2002** [comment]
Actually, mine was a question if this was somehow already implemented. So I assume it is not..:)

I'm afraid this will be quite complicated. As far as I know eclipse stores plugins and "core" components (which you would like to be updated by the new eclipse version) in the same folder.

Furthermore, plugins configuration data is stored across multipe directories: 
- user's workspace
- project settings
- configuration folder in the eclipse install directory

See here for a detailed explanation.

https://wiki.eclipse.org/FAQ_Where_do_plug-ins_store_their_state%3F

Is this a case where it would make sense to use "scoop hold" and use the eclipse auto installer to update the software? 

Edit: fixed typos

**Reply by rashil2000** [reply]
They still might have different subfolders, right? We can deal with that too. 

If you're using `scoop hold` to pause updates from Scoop's side, and use the builtin updater to update the IDE, then I'd suggest not using Scoop for installing Eclipse at all.

**Comment by finrod2002** [comment]
Maybe it would be enough to persist the "configuration" directory, e.g.:

scoop\apps\eclipse-jee\2023-06\configuration

Other relevant things should be persisted "automatically" as they are stored in the workspace and in the project folder, which are outside the eclipse installation folder.

**Reply by rashil2000** [reply]
> Maybe it would be enough to persist the "configuration" directory, e.g.:
> 
> scoop\apps\eclipse-jee\2023-06\configuration

Please raise a PR for this!
